lua/luasnip/nodes/snippet.lua

```diff
@@ -76,7 +64,25 @@ local function S(context, nodes, condition, ...)
 		active = false,
 		env = {},
 	})
+
 	snip:init_nodes()
+
+	if not snip.insert_nodes[0] then
+		-- Generate implied i(0)
+		local i0 = iNode.I(0)
+		nodes[#nodes + 1] = i0
+		i0.parent = snip
+		snip.insert_nodes[0] = i0
+
+		local last = #snip.insert_nodes
+		i0.indx = last + 1
+
+		if last ~= 0 then
+			i0.prev = snip.insert_nodes[last]
+			snip.insert_nodes[last].next = i0
+		end
```

Comment:
Looks good, that part is actually not necessary, `insert_nodes[0]` is not linked to the other insert_nodes (not really a fitting name anymore, should be eg. jumpables), it is jumped to after exiting the snippet (see lines 195 and 198).

Reply:
Ah sweet, much better then